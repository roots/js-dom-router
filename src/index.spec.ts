import Router from './index';

describe('DOM router', () => {
  function makeElement(className:string) : HTMLElement {
    const el = document.createElement('div');
    el.className = className;
    document.body.append(el);
    return el;
  }

  test('it should fire an event for each class name', () => {
    const el = makeElement('kjo apray');
    const callback = jest.fn(() => el.className);

    el.addEventListener('router.kjo', callback);
    el.addEventListener('router.apray', callback);

    new Router(el); // tslint:disable-line

    expect(callback).toMatchSnapshot();
  });

  test('it should use body classes when document is passed', () => {
    const callback = jest.fn(() => document.body.className);
    document.body.className = 'kjo apray';

    document.addEventListener('router.kjo', callback);
    document.addEventListener('router.apray', callback);

    new Router(document); // tslint:disable-line

    expect(callback).toMatchSnapshot();
  });

  test('it should watch target for new classes', (done) => {
    const el = makeElement('kjo');
    const callback = jest.fn(() => el.className);

    const router = new Router(el);

    router.watch();

    el.addEventListener('router.kjo', callback);
    el.addEventListener('router.apray', callback);
    el.addEventListener('router.bdubs', callback);

    setTimeout(() => el.className += ' apray', 100);
    setTimeout(() => el.className += ' budbs', 200);
    setTimeout(() => el.className = 'kjo swalk apray bdubs', 300);
    setTimeout(() => {
      expect(callback).toMatchSnapshot()
      done();
    }, 400);
  });
});
