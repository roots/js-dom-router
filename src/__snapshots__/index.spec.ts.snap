// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOM router it should fire an event for each class name 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Event {
        "isTrusted": false,
      },
    ],
    Array [
      Event {
        "isTrusted": false,
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": "kjo apray",
    },
    Object {
      "type": "return",
      "value": "kjo apray",
    },
  ],
}
`;

exports[`DOM router it should use body classes when document is passed 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Event {
        "isTrusted": false,
      },
    ],
    Array [
      Event {
        "isTrusted": false,
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": "kjo apray",
    },
    Object {
      "type": "return",
      "value": "kjo apray",
    },
  ],
}
`;

exports[`DOM router it should watch target for new classes 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Event {
        "isTrusted": false,
      },
    ],
    Array [
      Event {
        "isTrusted": false,
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": "kjo apray",
    },
    Object {
      "type": "return",
      "value": "kjo swalk apray bdubs",
    },
  ],
}
`;
